cmake/CheckTrezor.cmake

```diff
@@ -102,13 +113,14 @@ if(Protobuf_FOUND AND USE_DEVICE_TREZOR AND TREZOR_PYTHON)
         "${CMAKE_BINARY_DIR}/test-protobuf.pb.cc"
         "${CMAKE_CURRENT_LIST_DIR}/test-protobuf.cpp"
         CMAKE_FLAGS
+        CMAKE_EXE_LINKER_FLAGS ${CMAKE_TRY_COMPILE_LINKER_FLAGS}
         "-DINCLUDE_DIRECTORIES=${Protobuf_INCLUDE_DIR};${CMAKE_BINARY_DIR}"
-        "-DCMAKE_CXX_STANDARD=11"
-        LINK_LIBRARIES ${Protobuf_LIBRARY}
+        "-DCMAKE_CXX_STANDARD=14"
```

Comment:
What's even the point of the compile test? Seems it has been failing since forever, see here in latest master: https://github.com/monero-project/monero/actions/runs/5392471097/jobs/9790860150#step:5:145

If you remove this test we can use the latest version of protobuf again without issues.

Reply:
nope, it is not failing neither for me locally nor for GHA runners. If I remove it and keep latest protobuf, it fails for me locally. Try upgrading protobuf to 23 locally and remove protobuf@21. maybe you have linked still v21?

Reply:
if you fetch this PR branch and try compiling brew-installed protobuf v23 with c++14 it fails regardless of try_compile result. Does GHA runner give another result? I tried it in this PR, it failed.

Reply:
making it work with latest protobuf is effor worth of hours, IMO for another PR. I dont have time for this rn

Reply:
also ad failing compile test - this PR fixes it, if compile test fails, trezor support is not included, thats one of primary reasons for this PR

Reply:
I will try if I can fix it myself, I said this before but the problem is that this will break compilation and the brew file doesn't fix it, it will just cause conflicts with already installed version of protobuf -> support requests that I will have to deal with :/ I'm just unhappy with it currently but I can understand if you don't want to spend more time on it currently.

Reply:
OK so in the next PR? :) 
Brewfile should relink protobuf@21 to be the main lib, so when installed, compilation should pass.

Reply:
Also pls note that this PR makes Trezor support mandatory, if some check fails and Trezor support could not be enabled, build fails. This was not the case before. So if protobufv23 failed on try compile, Trezor support was disabled and compilation passed. We dont want this to happen anymore as official builds sometimes lacked Trezor support (it is still possible to explicitly disable trezor support via cmake option, but by default, it is enabled.)

Reply:
another approach IMO: a) c++17, then lets use new protobuf, b) allow using protobufv21 in not linked version, but then user should override protobuf version that is going to be used - env vars probably

Reply:
If I remember correctly we used C++14 because of using an old XCode version for macOS binaries. We updated the XCode version a year ago so it might be possible for us to switch to C++17. I'll look into it.

Reply:
This also works

```
CMAKE_PREFIX_PATH=/opt/homebrew/Cellar/protobuf@21/21.12 make debug-test-trezor -j8
```